File: 백준/Silver/15565. 귀여운 라이언/귀여운 라이언.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
N,K = map(int, input().split())
dolls = list(map(int, input().split()))
ryan = []
for i in range(0, N):
    if dolls[i]==1:
        ryan.append(i)
start, end = 0, K-1
min_length = 1000001
if len(ryan)<K:
    print(-1)
else:
    while start != end:
        ryan_count = ryan[end]-ryan[start]+1
        min_length = min(min_length, ryan_count)
        if end ==len(ryan)-1:
            break
        start+=1
        end+=1
    print(min_length)

File: 백준/Silver/15565. 귀여운 라이언/귀여운 라이언.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n, k, r int
	fmt.Fscanf(reader, "%d %d\n", &n, &k)

	var ryan = []int{}
	for i := 0; i < n; i++ {
		fmt.Fscanf(reader, "%d ", &r)
		if r == 1 {
			ryan = append(ryan, i)
		}
	}
	if len(ryan) < k {
		fmt.Fprintf(writer, "%d\n", -1)
	} else {
		var start = 0
		var end = k - 1
		var ryanCount = 0
		var min = 1000001
		for start != end {
			ryanCount = ryan[end] - ryan[start] + 1
			if ryanCount < min {
				min = ryanCount
			}
			if end == len(ryan)-1 {
				break
			}
			start++
			end++
		}
		fmt.Fprintf(writer, "%d\n", min)
	}
}
